JP-3669: 'OLS' Conditional Read Noise Recalculation for CHARGELOSS (#8697)
spacetelescope/jwst: CHANGES.rst

- Refactored separate modes into submodules instead of inheriting from a base class.
  Moved non-JWST-specific code to stcal. [#8613]

ramp_fitting
------------

- Updating the ramp fitting work flow for the read noise variance calculation
  for ramps containing CHARGELOSS flags.  When running the ``OLS_C`` ramp

>   fitting algorith, this recalculation is being done in the C-extension, so

Reviewer: ```suggestion
  fitting algorithm, this recalculation is being done in the C-extension, so
```

- Refactored separate modes into submodules instead of inheriting from a base class.
  Moved non-JWST-specific code to stcal. [#8613]

ramp_fitting
------------

> - The STCAL ramp fitting ``OLS_C`` algorithm has been modified to handle the
>   read noise variance recalculation for CHARGELOSS flagging.  Because of this
>   the JWST ramp fitting step code read noise recalculation should only be run
>   when the ``OLS`` algorithm, which runs the legacy python code for ramp
>   fitting.  Running the step code read noise recalculation when running the
>   ``OLS_C`` algorithm incorrectly runs the recalcuation twice.  Because of this
>   the JWST step code has been modified to account for the change in STCAL.
>   [#8697, spacetelescope/stcal#278]
> 

Reviewer 2: Thanks for the changes!  But I think this could still be more concise:

```suggestion
- Moved the read noise variance recalculation for CHARGELOSS flagging to the C
  implementation, when the algorithm is ``OLS_C``. [#8697, spacetelescope/stcal#278]

```

- Removed many non-working and out-dated scripts. Including
  many scripts that were replaced by ``strun``. [#8619]

set_telescope_pointing
----------------------

- replace usage of ``copy_arrays=True`` with ``memmap=False`` [#8660]

- Refactored separate modes into submodules instead of inheriting from a base class.
  Moved non-JWST-specific code to stcal. [#8613]

> 

Reviewer 2: Looking at #8613, I think this comment was supposed to go under the outlier detection heading.

Author: This isn't my change.  That bullet is under `set_telescope_pointing` in the master repo.  It only appears as my edit because of git's problem with CHANGES.rst always claiming merge conflicts:

https://github.com/spacetelescope/jwst/blob/9667e6c8fcdf316aa558544afa54c7b8528a5540/CHANGES.rst?plain=1#L70

Reviewer 2: Thanks, it should be fixed on main now.
